services/inc/atomic_flag_mutex.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <atomic>
+#include <mutex>
+
+template <typename R, R(*spin)()>
+class AtomicFlagMutex {
+	std::atomic_flag flag = ATOMIC_FLAG_INIT;
+public:
+	void lock() {
+	    while (flag.test_and_set(std::memory_order_acquire)) {
+	        spin();
+	    }
+	}
+
+	void unlock() {
+	    flag.clear(std::memory_order_release);
+	}
+};
```

Comment:
I would like to see this `AtomicFlagMutex` implementation supporting recursive locking, possibly in next versions of the firmware. For example:
```cpp
void lock() {
    acquire_flag();
    if (!mutex_is_initialized && mutex_can_be_initialized) { // Heap is initialized, etc.
        init_recursive_mutex();
        mutex_is_initialized = true;
    }
    if (mutex_is_initialized) {
        lock_mutex();
        release_flag();
    }
}

void unlock() {
    if (mutex_is_initialized) {
        unlock_mutex();
    } else {
        release_flag();
    }
}
```